misc/package/Data Files/MWSE/core/mcm/components/categories/Category.lua

```diff
@@ -45,6 +45,11 @@ function Category:new(data)
 	local parent = t.parentComponent
 	if not parent then return t end
 
+	if t.showDefaultSetting == nil then
+		-- Using `rawget` so we don't inherit a default value
+		t.showDefaultSetting = rawget(parent, "showDefaultSetting")
```

Comment:
I generally like the idea. But can you explain why you use rawget?

Reply:
I don't think it's necessary here, so I can remove it if you want. I mainly just used it out of some fear that maybe in the future the default of the `showDefaultSetting` field might be changed. e.g. maybe something like 

```lua
-- in Category.lua
Category.showDefaultSetting = false

-- in LogLevelOptions.lua
LogLevelOptions.showDefaultSetting = true
```
If `rawget` isn't used, then (in this made up hypothetical), newly constructed `LogLevelOptions` objects would inherit the `false` value from `Category` instead of the `true` value from `LogLevelOptions`.

But as long as `Category`-type things don't have a value for `showDefaultSetting` set at the class level, it should be fine to not use `rawget`.

Reply:
I see. Using `rawget` is fine.